### PR TITLE
[ci] Download Verilator from GCP

### PR DESCRIPTION
**Pull request by mundaym**

### CODE REVIEW

File: ci/install-package-dependencies.sh
Comment (drewmacrae) [inline ci/install-package-dependencies.sh:132]: As before, if we merge code that depends on a new version of verilator, we should ask others to rebase their PRs so they get the update in the version of azure-pipelines.yml that they use for their CI runs.

**Review by drewmacrae — APPROVED** [review]
I'd like for this PR's discussion to have a bit more on our motivation here before it gets merged, but it looks generally good. Why do we prefer a custom script+GCP to OBS+apt.

LGTM pending some sort of justification

**Comment by GregAC** [comment]
> Why do we prefer a custom script+GCP to OBS+apt.

I'd say 4 reasons:

1. OBS is a bit flaky, it's unreliability has caused a few CI outages (one of which was the motivation for this PR)
2. OBS recently identified our Verilator packages as not being actively updated so stated they would be removed unless we asked for them not to be. We of course asked for them to be kept but highlighted OBS policies on what they want to keep on their service aren't known to us (I did go looking a while ago and couldn't find any explicit terms), It's entirely possible they could decide our use case isn't something they support (in effect using it as a supporting part of our CI infrastructure) and remove the packages without warning.
3. We had a prior outage due to an expiring key for signing the packages, easy enough to fix (and we can ensure we setup an automated reminder to refresh signing keys) but another thing to trip up on and ensure we maintain correctly.
4. Pushing new Verilator versions is a little fiddly. It's not particular complex but with the GCP setup we can build binaries and upload them for any Verilator github commit with the push of a button.

(I wrote this, @mundaym picked it up as I had it out as a draft PR but wasn't available and we wanted to deploy ASAP to fix an OBS issue, that went away in the meantime)

**Comment by jprendes** [comment]
We will need this to unblock the Ubuntu update.

**Review by jprendes — COMMENTED** [review]
LGTM

File: ci/install-package-dependencies.sh
Comment (mundaym) [reply ci/install-package-dependencies.sh:128]: We're grabbing the binary from a different source but this PR doesn't change the version of Verilator that we are using so I think it shouldn't affect anyone using OT. We'll do an announcement when we update to v4.228.

**Comment by mundaym** [comment]
Thanks all, I'm going to merge this as the OBS build was flaky again today. It will also allow us to proceed with the change to Ubuntu 20.04 and the Verilator update to v4.228.

**Pull request merged**
